evm/rpc/main.py

```diff
@@ -0,0 +1,115 @@
+import json
+import logging
+
+from evm.exceptions import (
+    ValidationError,
+)
+
+from evm.rpc.modules import (
+    Eth,
+    EVM,
+)
+
+REQUIRED_REQUEST_KEYS = (
+    'id',
+    'jsonrpc',
+    'method',
+)
+
+
+def validate_request(request):
+    for key in REQUIRED_REQUEST_KEYS:
+        if key not in request:
+            raise ValueError("request must include the key %r" % key)
+
+
+def generate_response(request):
+    return {
+        'id': request.get('id', -1),
+        'jsonrpc': request.get('jsonrpc', "2.0"),
+    }
+
+
+class RPCServer:
+    '''
+    This "server" accepts json strings requests and returns the appropriate json string response,
+    meeting the protocol for JSON-RPC defined here: https://github.com/ethereum/wiki/wiki/JSON-RPC
+
+    The key entry point for all requests is :meth:`RPCServer.request`, which
+    then proxies to the appropriate method. For example, see
+    :meth:`RPCServer.eth_getBlockByHash`.
+    '''
+
+    module_classes = (
+        Eth,
+        EVM,
+    )
+
+    def __init__(self, chain):
+        self.modules = {}
+        self.chain = chain
+        for M in self.module_classes:
+            self.modules[M.__name__.lower()] = M(chain)
```

Comment:
*nitpick*

Enforcing uniqueness on the set of module names would save us a potentially confusing edge case.